### ISSUE TITLE
Make rule SC2178 work

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/shellcheck-repl#16]
Adopted from https://github.com/HenrikBengtsson/shellcheck-repl/issues/15#issuecomment-1043892879;

```sh
$ foo=(1 2 3); foo="abc"
             ^-^ SC2178 (warning): Variable was used as an array but is now assigned a string.
```

However, it's not detected with:

```sh
$ foo=(1 2 3)
$ foo="abc"
```

because we don't pick up `foo` as being preexisting in the second case.

**Comment by HenrikBengtsson** [comment]
The task here is to identify what environment variables are assign/modified.  If that can be done, then we can `declare -p` those variables and SC2178 checks will detect this problem.

**Comment by HenrikBengtsson** [comment]
I think we have leveraged SC2034 here, e.g.

```sh
input=$READLINE_LINE
# e.g.
input='a=(1 2 3); b="abc"'
```
which gives
```sh
$ shellcheck --shell=bash --format=gcc <(echo "$input") 
/dev/fd/63:1:1: warning: a appears unused. Verify use (or export if used externally). [SC2034]
/dev/fd/63:1:12: warning: b appears unused. Verify use (or export if used externally). [SC2034]

```

**Comment by HenrikBengtsson** [comment]
Implemented;

```sh
$ a=(1 2 3)
$ a="abc"
^-- SC2178 (warning): Variable was used as an array but is now assigned a string.
```
